tests/align/talign.nim

```diff
@@ -51,3 +51,26 @@ type Bug[T] = object
 
 var bug: Bug[int]
 doAssert sizeof(bug) == 128, "Oops my size is " & $sizeof(bug) # 16
+
+
+import std/asyncdispatch
```

Comment:
this doesn't need async - a normal closure iterator would exhibit the same bug

Reply:
```nim
type
  ValidatorPubKey = object
    blob: array[96, byte]

proc f(): auto =
  return iterator() =
    var pad: int8 = 0
    var y {.align: 16.}: ValidatorPubKey
    let value = cast[uint64](addr y)
    doAssert value mod 16 == 0

f()()
```

Reply:
thanks!